client/allocrunner/alloc_runner.go

```diff
@@ -877,36 +877,37 @@ func (ar *allocRunner) destroyImpl() {
 	ar.destroyedLock.Unlock()
 }
 
-func (ar *allocRunner) PersistState() error {
+func (ar *allocRunner) PersistState(inBatch bool) error {
```

Comment:
There's only one caller for `PersistState`, so this argument is always true. Let's remove this flag.